### ISSUE TITLE
can not upload attachment

### ISSUE THREAD

**Issue by myheren** [yogiben/meteor-admin#224]
I used meteor-starter project and in it I can not upload attachment of posts, as in the picture
![t1](https://cloud.githubusercontent.com/assets/2194048/8462413/5917cd4a-2066-11e5-8b60-1bcc04218687.JPG)

**Comment by reduxdj** [comment]
+1, looks like quickform is broken?

**Comment by mordra** [comment]
in your posts.coffee file:
https://github.com/yogiben/meteor-starter/blob/13efb700d9f8c417070234c54f868ab76fad8323/collections/posts.coffee

you need to add between line 27-28: 
`optional:true`

just like how the updatedAt field has it on line 21

**Comment by reduxdj** [comment]
I changed my model to Articles rather than posts:

That does not help. still does not work, nothing happens in the UI when i goto insert an article from /admin/Articles/new - image upload does not work from this template,

However, when I goto my custom /article/new template it works fine.  Can you fix the admin tool to make image upload work please?

**Comment by mpowaga** [comment]
I can confirm it doesn't work with 1.2.0. It does work with 1.2.1 however which should be released today.

**Comment by reduxdj** [comment]
That's reassuring

**Comment by mpowaga** [comment]
It's already on atmosphere. You can check it out.

**Comment by reduxdj** [comment]
1.2.1 still fails out of the box for me.

2http-call-client.js:101 The provided value 'undefined' is not a valid enum value of interface XMLHttpRequestResponseType.
fsFile-common.js:46 Uncaught Error: FS.File.attachData requires a data argument with some datafsFileAttachData @ fsFile-common.js:46FS.Collection.insert @ api.common.js:81Template.afFileUpload.events.change .js-file @ autoform-file.coffee:51(anonymous function) @ template.js:463Template._withTemplateInstanceFunc @ template.js:437(anonymous function) @ template.js:462(anonymous function) @ view.js:869Blaze._withCurrentView @ view.js:523(anonymous function) @ view.js:868(anonymous function) @ events.js:47jQuery.event.dispatch @ jquery.js:4665jQuery.event.add.elemData.handle @ jquery.js:4333

**Comment by reduxdj** [comment]
mm: this works (didn't get the latest version when I did a meteor remove then add, had to explicitly define in packages.
yogiben:admin@1.2.1
yogiben:pretty-email
yogiben:autoform-file
yogiben:autoform-map
yogiben:helpers

**Comment by reduxdj** [comment]
Please close this ticket.

**Comment by reduxdj** [comment]
I saw it work once, but it's still a problem, would you know why cfs_upload throws that error.

**Comment by reduxdj** [comment]
Even supplying my own template results in the same problem.. while that template works with upload by itself.

**Comment by reduxdj** [comment]
By the way I did a meteor remove and a meteor add to re-add the package, and 1.1.1 installed instead of 1.2.1. So I changed the package file and it updated, still I can't get image upload to work, I get a browser error. Again, If I go and create my own edit page, without the admin sidebar, the image upload works, when the form is nested in the admin view image upload does not work.

I'm hoping this gets resolved, I've setup a simple Meteor admin for a CMS for my client and they are doing a photography app, so image upload support on the admin is really a super big desire of mine!

Do you know when this will be updated to work and will the meteor starter app also be updated?

**Comment by reduxdj** [comment]
I don't think this is an issue of the library but an issue of the package manager, cleaning out the file, and rebuilding the whole thing fixed my issue and upload works.

https://github.com/meteor/meteor/issues/4746

**Comment by reduxdj** [comment]
I don't know what's happening, but the darn upload stopped working on meteor admin for me. I don't know why it's having a problem... it was working and now no image is uploaded.

**Comment by reduxdj** [comment]
The problem seems to be do a missing suscription in the meteor starter app:

Meteor.subscribe('attachments') solved the problem for me.

**Comment by jlourenco** [comment]
Yep, also solved for me...

Router.waitOn ->
  [
    subs.subscribe 'user'
    subs.subscribe 'attachments'
  ]

at router.coffee

**Comment by headkit** [comment]
Works in Chrome and Safari but still does not work in Firefox :-/
